File: Low Level Design (LLD)/DesignPatterns/1. Creational/Prototype/Cpp/main.cpp
// main.cpp
#include "TShirt.h"

int main() {
    TShirt* baseTShirt = new TShirt("M", "Black");

    TShirt* customer1Shirt = static_cast<TShirt*>(baseTShirt->clone());
    TShirt* customer2Shirt = static_cast<TShirt*>(baseTShirt->clone());

    customer1Shirt->setColor("Red");
    customer2Shirt->setSize("L");

    baseTShirt->showDetails();
    customer1Shirt->showDetails();
    customer2Shirt->showDetails();

    // Clean up
    delete baseTShirt;
    delete customer1Shirt;
    delete customer2Shirt;

    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/1. Creational/Prototype/Java/main.java
public class Main {
    public static void main(String[] args) {
        // Base product
        TShirt baseTShirt = new TShirt("M", "Black");

        // Clone the product
        TShirt customer1Shirt = (TShirt) baseTShirt.clone();
        TShirt customer2Shirt = (TShirt) baseTShirt.clone();

        // Customize clones
        customer1Shirt.setColor("Red");
        customer2Shirt.setSize("L");

        // Show details
        baseTShirt.showDetails();
        customer1Shirt.showDetails();
        customer2Shirt.showDetails();
    }
}
